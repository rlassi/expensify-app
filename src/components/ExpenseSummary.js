import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import expensesTotal from '../selectors/expenses-total';
import visibleExpenses from '../selectors/expenses'

export const ExpenseSummary = (props) => {
    const filteredExpenses = visibleExpenses(props.expenses, props.filters);
    const totalAmount = numeral(expensesTotal(filteredExpenses) / 100).format('$0,0.00')
    return (
        <div>
            {
                filteredExpenses.length === 1 ?
                    (
                        <h1>Viewing 1 expense totalling {totalAmount}</h1>
                    ) : (
                        <h1>Viewing {filteredExpenses.length} expenses totalling {totalAmount}</h1>
                    )
            }
        </div>
    )
};

const mapStateToProps = (state) => ({
    expenses: state.expenses,
    filters: state.filters
});

export default connect(mapStateToProps)(ExpenseSummary);
