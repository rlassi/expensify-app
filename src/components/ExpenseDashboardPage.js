import React from 'react';
import ConnectedExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseSummary />
        <ExpenseListFilters />
        <ConnectedExpenseList />
    </div>
);

export default ExpenseDashboardPage;