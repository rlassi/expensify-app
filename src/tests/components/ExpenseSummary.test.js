import React from 'react';
import { shallow } from 'enzyme';

import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';
import { filters } from '../fixtures/filters';

test('should render ExpenseSummary showing 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expenses={[ expenses[0] ]} filters={filters}/>);
    expect(wrapper).toMatchSnapshot();
})

test('should render ExpenseSummary showing 0 or multiple expenses', () => {
    let wrapper = shallow(<ExpenseSummary expenses={[]} filters={filters} />);
    expect(wrapper).toMatchSnapshot();

    wrapper = shallow(<ExpenseSummary expenses={expenses} filters={filters} />);
    expect(wrapper).toMatchSnapshot();
})
