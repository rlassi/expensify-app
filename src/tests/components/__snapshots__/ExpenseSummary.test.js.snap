// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary showing 0 or multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary showing 0 or multiple expenses 2`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary showing 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
